spec/std/enumerable_spec.cr

```diff
@@ -17,23 +38,27 @@ describe "Enumerable" do
     end
   end
 
-  describe "all? with block" do
-    it "returns true" do
-      ["ant", "bear", "cat"].all? { |word| word.length >= 3 }.should be_true
+  describe "drop_while" do
+    it "drops elements while the condition holds true" do
+      result = [1, 2, 3, 4, 5, 0].drop_while {|i| i < 3}
+      result.should eq [3, 4, 5, 0]
     end
 
-    it "returns false" do
-      ["ant", "bear", "cat"].all? { |word| word.length >= 4 }.should be_false
+    it "returns an empty array if the condition is always true" do
+      [1, 2, 3].drop_while {true}.should eq [] of Int32
     end
-  end
 
-  describe "all? without block" do
-    it "returns true" do
-      [15].all?.should be_true
+    it "returns the full Array if the the first check is false" do
+      [5, 0, 1, 2, 3].drop_while {|x| x < 4}.should eq [5, 0, 1, 2, 3]
     end
 
-    it "returns false" do
-      [nil, true, 99].all?.should be_false
+    it "does not yield to the block anymore once it returned false" do
```

Comment:
I see your intention. But I think call `drop_while` with block having side effect is not appropriate :(